File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def rec_func(index, lst):
            nonlocal res
            
            if index == len(nums):
                res.append(lst)
                return
            
            lst.append(nums[index])
            rec_func(index+1, lst.copy())
            lst.pop()
            rec_func(index+1, lst.copy())
            
        rec_func(0, [])
        return res

File: 0078-subsets/0078-subsets.go
var Result [][]int = [][]int{}

func subsets(nums []int) [][]int {
    Result = [][]int{}
    RecFunc(0,[]int{}, nums)
    return Result
}

func RecFunc(index int, subset []int,  nums []int) {
    if index == len(nums){
        copySubset := make([]int, len(subset))
        copy(copySubset, subset)
        Result = append(Result, copySubset)
        return
    }
    
    RecFunc(index+1, append(subset, nums[index]), nums)
    RecFunc(index+1, subset, nums)
}